dm/tests/safe_mode/run.sh

```diff
@@ -183,9 +183,93 @@ function safe_mode_recover() {
 	done
 }
 
+function safe_mode_duration() {
+	run_sql_file $cur/data/db1.prepare.sql $MYSQL_HOST1 $MYSQL_PORT1 $MYSQL_PASSWORD1
+	check_contains 'Query OK, 2 rows affected'
+	run_sql_file $cur/data/db2.prepare.sql $MYSQL_HOST2 $MYSQL_PORT2 $MYSQL_PASSWORD2
+	check_contains 'Query OK, 3 rows affected'
+
+	export GO_FAILPOINTS="github.com/pingcap/tiflow/dm/syncer/SafeModeDurationSetBeginLoc=return()"
+	run_dm_master $WORK_DIR/master $MASTER_PORT $cur/conf/dm-master.toml
+	check_rpc_alive $cur/../bin/check_master_online 127.0.0.1:$MASTER_PORT
+	# worker1 -> source1
+	run_dm_worker $WORK_DIR/worker1 $WORKER1_PORT $cur/conf/dm-worker1.toml
+	check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER1_PORT
+	cp $cur/conf/source1.yaml $WORK_DIR/source1.yaml
+	sed -i "/relay-binlog-name/i\relay-dir: $WORK_DIR/worker1/relay_log" $WORK_DIR/source1.yaml
+	dmctl_operate_source create $WORK_DIR/source1.yaml $SOURCE_ID1
+	# worker2 -> source2
+	run_dm_worker $WORK_DIR/worker2 $WORKER2_PORT $cur/conf/dm-worker2.toml
+	check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER2_PORT
+	cp $cur/conf/source2.yaml $WORK_DIR/source2.yaml
+	sed -i "/relay-binlog-name/i\relay-dir: $WORK_DIR/worker2/relay_log" $WORK_DIR/source2.yaml
+	dmctl_operate_source create $WORK_DIR/source2.yaml $SOURCE_ID2
+
+	dmctl_start_task "$cur/conf/dm-task-safe-mode-duration.yaml" "--remove-meta"
```

Comment:
failpoint `SafeModeDurationSetBeginLoc` makes tests can't reflect real scenario, the execution path is changed. You can inject some errors in `DMLWorker.executeBatchJobs`

Reply:
After https://github.com/pingcap/tiflow/pull/6225/commits/def5228201236d7ee3c226abbc6c1f9ae080d02a, the prepare sql can be executed to downstream. So needn't inject errors in `DMLWorker.executeBatchJobs`